lib/web_ui/lib/src/engine/skwasm/skwasm_impl/raw/raw_canvas.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@DefaultAsset('skwasm')
+library skwasm_impl;
+
+import 'dart:ffi';
+
+import 'package:ui/src/engine/skwasm/skwasm_impl.dart';
+
+class CanvasWrapper extends Opaque {}
+
+typedef CanvasHandle = Pointer<CanvasWrapper>;
+
+@Native<Void Function(CanvasHandle)>(symbol: 'canvas_destroy', isLeaf: true)
```

Comment:
The native engine annotations actually use a different convention in which they put the `@Native` annotations directly on methods of a dart class, and there is some sort of implicit lifetime linking between the C++ class and the dart class. I don't believe this convention works with the wasm implementation of the `@Native` annotation, so each of these entrypoints is a plain function. I felt it was nice to have each individual `raw_*.dart` file be a one-to-one mapping of the `.cpp` file in the skwasm target. For example, `raw_canvas.dart` corresponds to `canvas.cpp`, and only reflects the exact APIs that are exposed by that file.

To be honest, I don't actually like the "implicit lifetime linking between the C++ class and the dart class" convention anyway, since it relies on a bunch of magic handshakes between the C++ and dart class that are not well documented (both the dart class and the C++ class actually have to subclass some special classes like `NativeFieldWrapperClass1` and `DartWrappable`)